src/messages.js

```diff
@@ -362,4 +358,24 @@ export const sayWhoMadeMe = async (config) => {
     const maintainers = `I am also maintained by ${contributed}`;
 
     return `${created}. ${maintainers}.`;
+};
+
+/**
+ * @summary builds an "already a diamond" message
+ * @param {boolean} isModerator is user a current moderator
+ * @param {boolean} wasModerator was user a moderator
+ * @returns {string}
+ */
+export const sayDiamondAlready = (isModerator, wasModerator) => {
+
+    /**
+     * @type {[boolean, string][]}
+     */
+    const messageMap = [
+        [isModerator, `${getRandomOops()} you already have a diamond!`],
+        [wasModerator, `are you *really* sure you want to be a moderator again?`]
+    ];
+
+    const [, message] = messageMap.find(([condition]) => condition) || [];
```

Comment:
This is really clever!

Reply:
Thanks - this is a nice technique I picked up long ago for easy processing of <condition, handler/message/etc> maps. That said, it's a bit inefficient being an O(n) algorithm - a smarter way would probably be switching conditions to a map of <condition, handler>, and access `messageMap["true"]` in O(1)